pp_hot.c

```diff
@@ -3211,20 +3211,16 @@ PP(pp_match)
         EXTEND(SP, nparens + i);
         EXTEND_MORTAL(nparens + i);
         for (i = !i; i <= nparens; i++) {
-            if (LIKELY((RXp_OFFS(prog)[i].start != -1)
-                     && RXp_OFFS(prog)[i].end   != -1 ))
+            if (LIKELY(RXp_OFFS_VALID(prog,i)))
             {
-                const I32 len = RXp_OFFS(prog)[i].end - RXp_OFFS(prog)[i].start;
-                const char * const s = RXp_OFFS(prog)[i].start + truebase;
-                if (UNLIKELY(  RXp_OFFS(prog)[i].end   < 0
-                            || RXp_OFFS(prog)[i].start < 0
-                            || len < 0
-                            || len > strend - s)
+                const I32 len = RXp_OFFS_END(prog,i) - RXp_OFFS_START(prog,i);
+                const char * const s = RXp_OFFS_START(prog,i) + truebase;
+                if ( UNLIKELY( len < 0 || len > strend - s)
```

Comment:
This condition seems to have lost the logic relating to `start/end < 0`. Maybe it wanted a `RXp_OFFS_VALID()` test as well?

Reply:
Its validated just above, at line 3214.

Reply:
The previous code validated twice. :-)

Reply:
Ohrighr, yes there it is. :)

Reply:
In a follow up code ill simplify that to extract the start offset and end offset just once for that code. its a bit repetitive.